packages/design-system/src/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,22 @@
+import { Root } from '@radix-ui/react-tooltip';
+import * as React from 'react';
+import { TooltipArrow } from './tooltip-arrow';
+import { TooltipContent } from './tooltip-content';
+import { TooltipProvider } from './tooltip-provider';
+import { TooltipTrigger } from './tooltip-trigger';
+
+type RootProps = React.ComponentPropsWithoutRef<typeof Root>;
+
+export interface TooltipProps extends RootProps {}
+
+export const TooltipRoot: React.FC<Readonly<TooltipProps>> = ({
+  children,
+  ...props
+}) => <Root {...props}>{children}</Root>;
+
+export const Tooltip = Object.assign(TooltipRoot, {
+  Arrow: TooltipArrow,
+  Provider: TooltipProvider,
+  Content: TooltipContent,
+  Trigger: TooltipTrigger,
+});
```

Comment:
instead of object.assign let's export an object instead, it's generally better for TypeScript
also, we export the Root separately, it would be like so: 

```suggestion
export const Tooltip = {
  Root: TooltipRoot,
  Arrow: TooltipArrow,
  Provider: TooltipProvider,
  Content: TooltipContent,
  Trigger: TooltipTrigger,
};
```